wicker/core/storage.py

```diff
@@ -281,19 +332,159 @@ def get_column_concatenated_bytes_files_path(self, s3_prefix: bool = True, datas
+                / assets
+                    - files added by users
+                / partition_1.parquet
+                / partition_2.parquet
+                    - _l5ml_dataset_partition_metadata.json
+                    - _SUCCESS
+                    - part-0-attempt-1234.parquet
+                    - part-1-attempt-2345.parquet
+    """
+
+    def __init__(self, s3_root_path: Optional[str] = None) -> None:
+        """Init S3PathFactory.
+
+        Args:
+            s3_root_path (Optional[str], optional): path to s3 root path. Defaults to None.
+        """
+        s3_config = get_config().aws_s3_config
+        store_concatenated_bytes_files_in_dataset = s3_config.store_concatenated_bytes_files_in_dataset
+        s3_root_path = s3_root_path if s3_root_path is not None else s3_config.s3_datasets_path
+        # ignore type as we already handled none case above
+        super().__init__(s3_root_path, store_concatenated_bytes_files_in_dataset)  # type: ignore
+
+    def get_dataset_assets_path(self, dataset_id: DatasetID, s3_prefix: bool = True) -> str:
+        """Get path to data assets folder.
+
+        Public gettr for data asset folder path logic.
+
+        Args:
+            dataset_id (DatasetID): Id to gather file path.
+            s3_prefix (bool, optional): Whether to keep the s3 prefix or not. Defaults to True.
+
+        Returns:
+            str: Path to data assets folder.
+        """
+        prefix = "s3://" if not s3_prefix else None
+        return self._get_dataset_assets_path(dataset_id=dataset_id, prefix=prefix)
+
+    def get_dataset_partition_metadata_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+        """Get metadata file path for partition.
+
+        Args:
+            data_partition (DatasetPartition): Partition to gather file path.
+            s3_prefix (bool, optional): Whether to keep the s3 prefix or not. Defaults to True.
+
+        Returns:
+            str: Path to dataset partition metadata file.
+        """
+        prefix = "s3://" if not s3_prefix else None
+        return self._get_dataset_partition_metadata_path(data_partition, prefix)
+
+    def get_dataset_partition_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+        """Get path to dataset partition data file.
+
+        Args:
+            data_partition (DatasetPartition): Partition to gather file path.
+            s3_prefix (bool, optional): Whether to keep the s3 prefix or not. Defaults to True.
+
+        Returns:
+            str: Path to dataset partition data file.
+        """
+        prefix = "s3://" if not s3_prefix else None
+        return self._get_dataset_partition_path(data_partition=data_partition, prefix=prefix)
+
+    def get_dataset_schema_path(self, dataset_id: DatasetID, s3_prefix: bool = True) -> str:
+        """Get path to the dataset schema.
+
+        Args:
+            dataset_id (DatasetID): Id of the dataset.
```

Comment:
Nit: consistent casing `Id` -> `ID`.

```suggestion
            dataset_id (DatasetID): ID of the dataset.
```